lib/bolt/cli.rb

```diff
@@ -362,6 +379,32 @@ def run_plan(plan_name, plan_arguments, nodes, options)
       result.ok? ? 0 : 1
     end
 
+    def apply_manifest(code, targets, filename = nil)
+      require 'bolt/catalog'
+      begin
+        ast = Puppet::Pops::Parser::EvaluatingParser.new.parse_string(code, filename)
+        ast_data = Puppet::Pops::Serialization::ToDataConverter.convert(ast, rich_data: true, symbol_to_string: true)
+      rescue Puppet::Error => e
+        raise Bolt::PAL::PALError, "Failed to parse manifest: #{e}"
```

Comment:
Is this really a PALError? We're not using PAL here.

Reply:
I used it because it's the closest thing we have to "an error from puppet". We have `Bolt::PuppetError` but that's meant to be literally a `Puppet::DataTypes::Error` object turned into a Bolt exception...